parser/scraper_halooglasi.py

```diff
@@ -0,0 +1,58 @@
+from bs4 import BeautifulSoup
+from parser.abstract_scraper import AbstractScraper
+from typing import Optional
+import parser_exceptions
+import logging
+
+
+class HalooglasiScraper(AbstractScraper):
+    BASE_URL = 'https://www.halooglasi.com/nekretnine/izdavanje-stanova'
+
+    def fetch_apartment_listings(self, page_content) -> list[AbstractScraper.ApartmentPageData]:
+        soup = BeautifulSoup(page_content.content, 'lxml')
+        return soup.find_all('div', class_='col-md-12 col-sm-12 col-xs-12 col-lg-12')
+
+    def get_apartment_title(self, apartment) -> Optional[str]:
+        product_title = apartment.find('h3', class_='product-title')
+        return product_title.find('a').text.strip()
+
+    def get_apartment_link(self, apartment) -> Optional[str]:
+        product_title = apartment.find('h3', class_='product-title')
+        return product_title.find('a')['href']
+
+    def get_rooms_number(self, apartment) -> Optional[str]:
+        try:
+            product_features = apartment.find('ul', class_='product-features')
+            features_items = product_features.find_all('li', class_='col-p-1-3')
+            rooms_div = features_items[1].find('div', class_='value-wrapper')
+            return rooms_div.get_text().replace('Broj soba', '')[:3]
+        except parser_exceptions.ApartmentRoomsNotFoundException:
```

Comment:
As i mentioned, use a single class error everywhere. And also please make a decorator that would execute a function and in case there is an error, it writes a log message and returns None. Name it `supress`. Decorator should take two parameters - Exception class and message.

```python
def supress(exception: Exception, error_msg: str):
....
```